lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -816,6 +816,26 @@ func (c *Client) sharedDirectoryMoveRequest(req tdp.SharedDirectoryMoveRequest)
 // the TDP connection to the browser.
 func (c *Client) close() {
 	c.closeOnce.Do(func() {
+		// In the case that the session ends due to an RDP server disconnect,
+		// the TDP connection will still be open at this point. Therefore
+		// we can ask for the RDP server disconnect reason here and send it
+		// back to the user via TDP.
+		res := C.get_server_disconnect_reason(c.rustClient)
+		if res.err != C.ErrCodeSuccess {
+			c.cfg.Log.Errorf("error getting server disconnect reason: %v", res.err)
+		} else {
```

Comment:
nit:

```suggestion
		if res.err != C.ErrCodeSuccess {
			c.cfg.Log.Errorf("error getting server disconnect reason: %v", res.err)
			return
		}
```

and ditch the indent, so it's a bit easier to read?

https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

Reply:
The issue with that in this context is that even if `C.get_server_disconnect_reason(c.rustClient)` errors, we still want to call the rest of the function. I'm using the same pattern with the calls below as well

```go
                 // Ensure the RDP connection is closed
		if errCode := C.close_rdp(c.rustClient); errCode != C.ErrCodeSuccess {
			c.cfg.Log.Warningf("error closing the RDP connection")
		} else {
			c.cfg.Log.Debug("RDP connection closed successfully")
		}

		// Ensure the TDP connection is closed
		if err := c.cfg.Conn.Close(); err != nil {
			c.cfg.Log.Warningf("error closing the TDP connection: %v", err)
		} else {
			c.cfg.Log.Debug("TDP connection closed successfully")
		}
```